pytest.ini

```diff
@@ -0,0 +1,11 @@
+[pytest]
+# Configure pytest to fail any test that raises an unexpected warning.
+# This will enable us to catch usages of deprecated features in downstream libraries,
+# before they are hard-deprecated.
+# Furthermore, they will catch any test issues which may hide bugs.
+# The format is `action:message_regex:category:module:line`.
+filterwarnings =
```

Comment:
TODO(cade): if the build passes with this, then we can merge. If not, then need to remove the pytest config and just merge the docstring changes